saleor/warehouse/availability.py

```diff
@@ -273,34 +290,145 @@ def check_preorder_threshold_bulk(
         )
         for variant_id, channel_listings in variant_channels.items()
     }
+    return (
+        variants_channel_availability,
+        variants_global_allocations,
+        all_variants_channel_listings,
+        variant_channels,
+    )
+
+
+def check_preorder_threshold_in_orders(
+    variant: "ProductVariant",
+    quantity: int,
+    channel_slug: str,
+    checkout_lines: Optional[Iterable["CheckoutLine"]],
+    check_reservations: bool,
+):
+    """Validate if there is preorder available for given variants in given country.
+
+    It is used in orders, since it does not need additional logic related to limits.
+    :raises InsufficientStock: when there is not enough items in stock for a variant.
+    """
+    (
+        variants_channel_availability,
+        variants_global_allocations,
+        all_variants_channel_listings,
+        variant_channels,
+    ) = get_channel_data([variant], channel_slug)
 
     if check_reservations:
-        quantity_reservation_list = (
-            PreorderReservation.objects.filter(
-                product_variant_channel_listing__in=all_variants_channel_listings,
-                quantity_reserved__gt=0,
+        listings_reservations = _check_reservations(
+            checkout_lines, all_variants_channel_listings
+        )
+    else:
+        listings_reservations = defaultdict(int)
+
+    insufficient_stocks: List[InsufficientStockData] = []
+
+    if variants_channel_availability[variant.id][1] is not None:
```

Comment:
IMO, data for given variant also should be named tuple as we totally do not one what is `variants_channel_availability[variant.id][1]`.

Reply:
Ofc i will fix that because you're right, but please note, that I just rearranged that part of code. Here is a proof:
https://github.com/saleor/saleor/blob/18a2f11f6217766fad8678b549fdaaba4099b608/saleor/warehouse/availability.py#L298 (master branch)

Reply:
Done, please also refer to https://github.com/saleor/saleor/pull/8405#discussion_r762126664